packages/shared/lib/participation/staking.ts

```diff
@@ -370,7 +379,107 @@ export const getTimeUntilMinimumAirdropReward = (account: WalletAccount, format:
     if (!account) return format ? getBestTimeDuration(0) : 0
 
     const [minRewards, minAirdrop, amountStaked] = getMinimumAirdropRewardInfo(account)
-    const remainingTime = calculateTimeUntilMinimumReward(minRewards, minAirdrop, amountStaked)
+    const timeRequired = calculateTimeUntilMinimumReward(minRewards, minAirdrop, amountStaked)
+
+    return format ? getBestTimeDuration(timeRequired) : timeRequired
+}
+
+const getNumRemainingMilestones = (airdrop: StakingAirdrop): number => {
+    const event = getStakingEventFromAirdrop(airdrop)
+    if (!event || !isStakingPossible(event?.status?.status)) return 0
+
+    const timeLeft =
+        airdrop === StakingAirdrop.Assembly
+            ? get(assemblyStakingRemainingTime)
+            : airdrop === StakingAirdrop.Shimmer
+            ? get(shimmerStakingRemainingTime)
+            : 0
+
+    const isInHolding = event?.status?.status === ParticipationEventState.Holding
+    const { milestoneIndexStart, milestoneIndexEnd } = event?.information
+
+    return isInHolding ? timeLeft / SECONDS_PER_MILESTONE : milestoneIndexEnd - milestoneIndexStart
+}
+
+const calculateIotasUntilMinimumReward = (rewards: number, airdrop: StakingAirdrop): number => {
+    const minRewardsRequired = getMinimumRewardsRequired(rewards, airdrop)
+    const numRemainingMilestones = getNumRemainingMilestones(airdrop)
+
+    return (minRewardsRequired * 1_000_000) / (numRemainingMilestones * getAirdropRewardMultipler(airdrop))
+}
+
+/**
+ * Calculates the remaining number of IOTAs needed to reach the minimum airdrop amount.
+ * If called with format = true then returns best unit match for amount of IOTAs, else
+ * returns the raw number of IOTAs.
+ *
+ * @method getIotasUntilMinimumAirdropReward
+ *
+ * @param {WalletAccount} account
+ * @param {boolean} format
+ *
+ * @returns {number | string}
+ */
+export const getIotasUntilMinimumAirdropReward = (account: WalletAccount, format: boolean = false): number | string => {
+    if (!account) return format ? formatUnitBestMatch(0) : 0
+
+    const [minRewards, minAirdrop, amountStaked] = getMinimumAirdropRewardInfo(account)
+    const iotasRequired = calculateIotasUntilMinimumReward(minRewards, minAirdrop)
+
+    return format ? formatUnitBestMatch(iotasRequired) : iotasRequired
+}
+
+/**
+ * Determines whether an account will be able to reach rewards minimums
+ * for both airdrops. This will return false if only one airdrop's minimum
+ * is able to be reached.
+ *
+ * @method canAccountReachMinimumAirdrop
+ *
+ * @param {WalletAccount} account
+ *
+ * @returns {boolean}
+ */
+export const canAccountReachMinimumAirdrop = (account: WalletAccount): boolean => {
+    if (!account) return false
+
+    const [minRewards, minAirdrop, amountStaked] = getMinimumAirdropRewardInfo(account)
```

Comment:
Using the current implementation of getMinimumAirdropRewardInfo, this will return [0, undefined, 0] which will intern cause this function to return 0. This is because you inside of getMinimumAirdropRewardInfo you only get the information if the account has or is participating in an airdrop.

This will then lead this function to return true when calculating if an account can reach the minimum airdrop before they ever stake it. Making the warning tooltip on the ManageStaking never show. (well once you fix my previous comment)